submissions/Oleksii_Drohachov/OOP-exercise/index.js

```diff
@@ -0,0 +1,98 @@
+class Creature {
+  constructor(name, gender, saying, friends) {
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+    this.propsArray = [
+      "species",
+      "name",
+      "gender",
+      "saying",
+      "friends",
+      "legs",
```

Comment:
A base class should never handle child classes' properties. This breaks Encapsulation, Open-Closed and Separation of Concerns principles as any changes to child classes would require changes in one (or more) base class up the hierarchy.

Look into method overloading to have consistent interface and user experience where method implementation depends on class yet allows to re-use parent class' method for a similar purpose.